sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/utils-service/_tests_/unit/application-event-code-during-enrolment-and-completed-utils.service.applicationEventCodeDuringCompleted.spec.ts

```diff
@@ -0,0 +1,219 @@
+    };
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it(`Should return ${ApplicationEventCode.DISC} when the disbursement schedule status is ${DisbursementScheduleStatus.Cancelled}.`, async () => {
+    // Arrange and act.
+    const applicationEventCode =
+      await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+        {
+          ...payload,
+          disbursementScheduleStatus: DisbursementScheduleStatus.Cancelled,
+        },
+      );
+    // Assert.
+    expect(applicationEventCode).toBe(ApplicationEventCode.DISC);
+  });
+
+  it(
+    `Should return ${ApplicationEventCode.COEA} when the disbursement schedule status is ${DisbursementScheduleStatus.Pending}, ` +
+      `COE status is ${COEStatus.completed} and today is before the cutoff date(i.e disbursement date + ${DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS}).`,
+    async () => {
+      // Arrange and act.
+      const applicationEventCode =
+        await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+          {
+            ...payload,
+            disbursementDate: formatDate(
+              addDays(
+                DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS + 1,
+                new Date(),
+              ),
+              DATE_ONLY_ISO_FORMAT,
+            ),
+          },
+        );
+      // Assert.
+      expect(applicationEventCode).toBe(ApplicationEventCode.COEA);
+    },
+  );
+
+  it(
+    `Should return ${ApplicationEventCode.DISR} when the disbursement schedule status is ${DisbursementScheduleStatus.Pending}, ` +
+      `COE status is ${COEStatus.completed}, disbursement date is same or after the cutoff date(i.e disbursement date + ${DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS}) ` +
+      `and has atleast one ${RestrictionActionType.StopFullTimeDisbursement}.`,
+    async () => {
+      // Arrange and act.
+      const applicationEventCode =
+        await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+          payload,
+          [[RestrictionActionType.StopFullTimeDisbursement]],
+        );
+      // Assert.
+      expect(applicationEventCode).toBe(ApplicationEventCode.DISR);
+    },
+  );
+
+  it(
+    `Should return ${ApplicationEventCode.COEA} when the disbursement schedule status is ${DisbursementScheduleStatus.Pending}, ` +
+      `COE status is ${COEStatus.completed}, disbursement date is same or after the cutoff date(i.e disbursement date + ${DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS}) ` +
+      `and does not have any ${RestrictionActionType.StopFullTimeDisbursement}.`,
+    async () => {
+      // Arrange and act.
+      const applicationEventCode =
+        await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+          payload,
+          [[RestrictionActionType.StopFullTimeApply]],
```

Comment:
is it not `RestrictionActionType.StopFullTimeDisbursement` ? comment also says the same

Reply:
Its the false case (i.e `does not have any ${RestrictionActionType.StopFullTimeDisbursement}`)
![image](https://github.com/bcgov/SIMS/assets/77353155/d9ccd655-7d4e-424a-901b-3fda2ec373d7)